### PR TITLE
KWS-2808: Refactor CreateUserNameView to the new iOS MVVM approach

### PR DESCRIPTION
**Pull request by Tbruni85**
## JIRA
https://jira.superawesome.tv/browse/KWS-2808    

## Description
Refactored the *CreateUserNameView* to updated MVVM solution discussed for iOS.
Events added:
* signup_start
* areaViewed

## Screenshots
n/a

### CODE REVIEW

File: PopJamUI/Classes/CreateUserNameView/Views/CreateUserNameView.swift [hunk 1]
```diff
@@ -255,7 +255,7 @@ public class CreateUserNameView: UIView, CreateUserNameDataProviderDelegate {
     private func verificationTextTimerAction(timer: Timer) {
         guard !userNametextView.isEmpty else { return }
         loadingState(isLoading: true)
-        dataProvider.verifyUsername(userName: userNametextView.textField.text!)
+        viewModel.verifyUserName(userName: userNametextView.textField.text!)
```
Comment (me2k) [inline PopJamUI/Classes/CreateUserNameView/Views/CreateUserNameView.swift]: `userNametextView` not camel case. replace with: `userNameTextView`

File: PopJamUI/Classes/CreateUserNameView/ViewModel/CreateUserNameViewModel.swift [hunk 2]
```diff
@@ -81,6 +110,40 @@ public struct CreateUserNameViewModel {
...
+        dataProvider.sendEvent(event: eventPayload)
+    }
+    
+    // MARK: CreateUserNameDataProvider Delegate methods
+    public func didGenerateRandomUserName(username: String) {
+        delegate?.didGenerateRandomUserName(username: username)
+    }
+    
+    public func usernameCanBeVerified(error: Error?) {
+        self.delegate?.usernameCanBeVerified(error: error)
```
Comment (me2k) [inline PopJamUI/Classes/CreateUserNameView/ViewModel/CreateUserNameViewModel.swift]: remove the `self.`

File: PopJamUI/Classes/CreateUserNameView/ViewModel/CreateUserNameViewModel.swift [hunk 3]
```diff
@@ -81,6 +110,40 @@ public struct CreateUserNameViewModel {
         ]
         let eventPayload = Stat.init(session: EventManger.shared.session, events: event)
         
-        //dataProvider.sendEvent(event: eventPayload)
+        dataProvider.sendEvent(event: eventPayload)
+    }
+    
+    public func sentAreaViewedEvent() {
+        
+        let event = [
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/CreateUserNameView/ViewModel/CreateUserNameViewModel.swift]: Wouldn't all these types of event definitions better be put in a common `EventFactory` or `StatsFactory` or something like that?

Then, if an `event name` changes, for example, you can update it once and have it available in all your data providers.

Comment (Tbruni85) [reply PopJamUI/Classes/CreateUserNameView/ViewModel/CreateUserNameViewModel.swift]: good point!
I'll update the EventManager with it once I've done with the refactor

**Pull request merged**